tests/test_views.py

```diff
@@ -0,0 +1,174 @@
+    for record in data:
+        cursor.execute(
+            "INSERT INTO api.events (event) VALUES (%s);",
+            (json.dumps(record),)
+        )
+
+def query_events_view(cursor):
+    cursor.execute("""
+        SELECT schema_version, occurred_at, user_id, event_type
+          FROM api.events_view
+      ORDER BY occurred_at;
+    """)
+    return cursor.fetchall()
+
+def load_expected(path):
+    """Return list of tuples (schema_version, occurred_at_str, user_id, event_type)."""
+    with open(path, newline='') as f:
+        reader = csv.DictReader(f)
+        rows = []
+        for row in reader:
+            rows.append((
+                row['schema_version'],
+                row['occurred_at'],
+                row['user_id'],
+                row['event_type'],
+            ))
+    return rows
+
+def format_timestamp(dt):
+    """Render a datetime as 'YYYY-MM-DD HH:MM:SS' (no tz offset)."""
+    return dt.strftime('%Y-%m-%d %H:%M:%S')
+
+def normalize_results(rows):
+    """
+    Given an iterable of (schema, datetime, user_id, event_type),
+    return a list of tuples where the datetime is formatted as
+    'YYYY-MM-DD HH:MM:SS' plus '+00' only for 'ltt' rows.
+    """
+    out = []
+    for schema, dt, user, ev in rows:
+        ts = dt.strftime('%Y-%m-%d %H:%M:%S')
+        if schema != 'legacy':
+            ts += '+00'
+        out.append((schema, ts, user, ev))
+    return out
+
+def run_test(mock_data, expected_rows):
+    """Wipe, insert mock_data, run view, normalize & assert against expected_rows."""
+    with psycopg2.connect(DATABASE_URL) as conn, conn.cursor() as cur:
+        insert_mock_data(cur, mock_data)
+        conn.commit()
+        raw = query_events_view(cur)
+    actual = normalize_results(raw)
+    assert actual == expected_rows
+
+@pytest.fixture
+def isacc_data():
+    path = os.path.join(DATA_DIR, "isacc_sample.json")
+    with open(path, 'r') as f:
+        return json.load(f)
+
+@pytest.fixture
+def isacc_expected():
+    return [
+        (
+            "3.0",                             # version → schema_version
+            "2025-04-23 14:22:10+00",          # occurred_at
+            "Practitioner/ABC-123",            # user_id (from subject)
+            "login"                            # event_type
+        )
+    ]
+
+@pytest.fixture
+def dhair2_data():
+    path = os.path.join(DATA_DIR, "dhair2_inform_sample.json")
+    return json.load(open(path))
+
+@pytest.fixture
+def dhair2_expected():
+    return [
+        (
+            "3.0",                      # version field
+            "2025-01-31 00:44:18+00",   # parsed from occurred
+            "patients/607",             # first item from subject array
+            "create"                    # action
+        )
+    ]
+
+@pytest.fixture
+def cosri_v2_data():
+    path = os.path.join(DATA_DIR, "cosri_v2_sample.json")
+    return json.load(open(path))
+
+@pytest.fixture
+def cosri_v2_expected():
```

Comment:
this probably doesn't need to be a fixture since it's only used in a single test